vision_datasets/common/data_manifest.py

```diff
@@ -324,6 +324,42 @@ def sample_subset_by_ratio(self, sampling_ratio):
         sampled_images = [self.images[i] for i in sampled_image_ids]
         return DatasetManifest(sampled_images, self.labelmap, self.data_type)
 
+    def sample_few_shots_subset_greedy(self, n_shots, random_seed=0):
+        """Greedy few-shots sampling method.
+        Randomly pick images from the original datasets until all classes have at least n_shots images.
+
+        Args:
+            n_shots (int): The number of images per class.
+            random_seed (int): Random seed to use.
+
+        Returns:
+            A samped dataset (DatasetManifest)
+
+        Raises:
+            RuntimeError if it couldn't find n_shots images.
+        """
+        assert n_shots > 0
+        images = list(self.images)
+        rng = random.Random(random_seed)
+        rng.shuffle(images)
+
+        num_classes = len(self.labelmap) if not self.is_multitask else sum(len(x) for x in self.labelmap.values())
+        total_counter = collections.Counter({i: n_shots for i in range(num_classes)})
```

Comment:
This does not take negative images into consideration I suppose

Reply:
Right, added a comment about negative images.